tracee-ebpf/tracee/events_pipeline.go

```diff
@@ -38,6 +44,79 @@ func (t *Tracee) handleEvents(ctx gocontext.Context) {
 	t.WaitForPipeline(errcList...)
 }
 
+// Under some circumstances, tracee-rules might be slower to consume events
+// than tracee-ebpf is capable of generating them. This requires
+// tracee-ebpf to deal with this possible lag, but, at the same,
+// perf-buffer consumption can't be left behind (or important events coming
+// from the kernel might be loss, causing detection misses).
+//
+// There are 3 variables connected to this issue:
+//
+// 1) perf buffer could be increased to hold very big amount of memory
+//    pages: The problem with this approach is that the requested space,
+//    to perf-buffer, through libbpf, has to be contiguous and it is almost
+//    impossible to get very big contiguous allocations through mmap after
+//    a node is running for some time.
+//
+// 2) raising the events channel buffer to hold a very big amount of
+//    events: The problem with this approach is that the overhead of
+//    dealing with that amount of buffers, in a golang channel, causes
+//    event losses as well. It means this is not enough to relief the
+//    pressure from kernel events into perf-buffer.
+//
+// 3) create an internal, to tracee-ebpf, buffer based on the node size.
+
+// queueEvents implements an internal FIFO queue for caching events
+func (t *Tracee) queueEvents(ctx gocontext.Context, in <-chan *external.Event) (chan *external.Event, chan error) {
+	out := make(chan *external.Event, 10000)
+	errc := make(chan error, 1)
+	done := make(chan bool, 1)
+
+	// idea here is to have multiple queue backends. for now we currently have
+	// a single backend keeping events in memory based on config events cache
+	// size (in MBs for this backend)
+	evtQueue := queue.EventQueueMem{}
```

Comment:
Could we pass this through tracee's `Config`? This way as a library consumer it would be easier to switch a backend, currently there would still be a need to manually change tracee's code.

So the config would have a field like this:
```
type Config struct {
    ...
    EventQueue EventQueue
}
```

Reply:
Done.